### ISSUE TITLE
Why PacketReceived event don't fire?

### ISSUE THREAD

**Issue by LJSJ123** [emancipatedMind/XmodemProtocol#9]
Hi, I asked some question on stackoverflow. 
Here is the [link](https://stackoverflow.com/questions/69736082/how-to-have-link-speed-and-transfer-percentage-when-sending-data-to-serial-port).
In question's update 2, I used 
``
xmodem.PacketReceived += (s, e) =>{
      Console.WriteLine(e.PacketReceived.Count);                        
};
``
to show the packet size on console.
But it doesn't show up.
Why PacketReceived event don't fire?

**Comment by emancipatedMind** [comment]
I have reviewed your code, and can only find one use for the communicator. That is when you are sending. If you are sending, you are not receiving. If you are not receiving, the PacketReceived event does not fire.

PacketReceived => You have received a packet.
PacketToSend => You are about to send a packet.

When sending, you are not looking to receive a packet. You are looking for an ACK, or NAK.
When receiving, you are not sending packets. You are only sending ACK or NAK.

I hope I answered your question.
